src/hooks/useAuth.tsx

```diff
@@ -157,10 +158,19 @@ export const AuthContextProvider = ({
     }
   }, [currentAppState, refreshIfNeeded]);
 
-  const refreshForAuthFailure = useCallback(async () => {
-    // TODO: Add throttling, etc.
-    return refreshIfNeeded(true);
-  }, [refreshIfNeeded]);
+  const refreshForAuthFailure = useCallback(
+    async (error: AxiosError) => {
+      const refreshTokenEndpointBlacklist = ['device-endpoints'];
+      if (
+        refreshTokenEndpointBlacklist.every(
+          (endpoint) => !error?.config?.url?.includes(endpoint),
```

Comment:
Device endpoints will fail if we attempt to re-register the same device with different attributes and cause an unnecessary token refresh.

Reply:
What are the different attributes? I would think we would want to try to reregister for a different account so we can get notifications for all user accounts.

Reply:
Responded offline, all set